### ISSUE TITLE
🔗 🐞 Bug report: Elementor 3.6.2 gap issue (Duplicate of #18309)

### ISSUE THREAD

**Issue by s5t5n** [elementor/elementor#18364]
### Prerequisites

- [X] I have searched for similar issues in both open and closed tickets and cannot find a duplicate.
- [X] The issue still exists against the latest stable version of Elementor.

### Description

Setting a gap brokes a grid. Containers inside main container get a wrong width, then a gap is used. When the main container settings are `--flex-direction: row; --flex-wrap: wrap;` setting a gap causes containers to get a wrong width which brokes a grid.

### Steps to reproduce

https://user-images.githubusercontent.com/29761044/161821957-3a2b139f-5b29-4a8b-8dac-2328f1a074fe.mp4

### Isolating the problem

- [X] This bug happens with only Elementor plugin active (and Elementor Pro).
- [X] This bug happens with a Blank WordPress theme active ([Hello theme](https://wordpress.org/themes/hello-elementor/)).
- [X] I can reproduce this bug consistently using the steps above.

### System Info

<html>
<body>

Server Environment |   |  
-- | -- | --
Operating System: | Linux |  
Software: | Apache/2.4.6 (CentOS) OpenSSL/1.0.2k-fips mod_fcgid/2.3.9 Phusion_Passenger/5.3.7 |  
MySQL version: | Percona Server (GPL), Release 30, Revision 8916819 v5.7.27-30 |  
PHP Version: | 7.4.27 |  
PHP Memory Limit: | 1024M |  
PHP Max Input Vars: | 10000 |  
PHP Max Post Size: | 128M |  
GD Installed: | Yes |  
ZIP Installed: | Yes |  
Write Permissions: | All right |  
Elementor Library: | Connected

</body>
</html>

**Comment by s5t5n** [comment]
I mean the width of containers should be something like `calc(container_width - var(--gap)` and last container should have 100% width.. hope you understand.

Last week i've use a code like this for jetwoobuilder products grid which have no gap property, only paddings and margin:

selector .jet-woo-products {
  gap: var(--gap);
  --gap: 40px;
}

selector .jet-woo-products__item {
    width: 100%;
    max-width: calc((100% / var(--columns)) - var(--gap) + (var(--gap) / var(--columns)));
}
`

**Comment by perks98** [comment]
I had reported this issue primarily in the beta stage but it doesn't appear it will get sorted any time soon. I think it needs more of the community to ask for this fix. They want you to add padding to the containers individually rather than use the gap or margin for grid layouts. If you check the gap issue I submitted #18309 and use roughly that CSS in the thread and use the flex-grow feature it should work how you want it roughly
